examples/src/main/scala/caliban/http4s/ExampleApp.scala

```diff
@@ -38,17 +42,23 @@ object ExampleApp extends CatsApp with GenericSchema[Console with Clock] {
   implicit val characterArgsSchema  = gen[CharacterArgs]
   implicit val charactersArgsSchema = gen[CharactersArgs]
 
-  def makeApi(service: ExampleService): GraphQL[Console with Clock] =
-    maxDepth(30)(
-      maxFields(200)(
-        graphQL(
-          RootResolver(
-            Queries(
-              args => service.getCharacters(args.origin),
-              args => service.findCharacter(args.name)
-            ),
-            Mutations(args => service.deleteCharacter(args.name)),
-            Subscriptions(service.deletedEvents)
+  def makeApi(service: ExampleService): UIO[GraphQL[Console with Clock]] =
+    apolloTracing(                  // wrapper for https://github.com/apollographql/apollo-tracing
+      printSlowQueries(500 millis)( // wrapper that logs slow queries
+        timeout(3 seconds)(         // wrapper that fails slow queries
+          maxDepth(30)(             // query analyzer that limit query depth
+            maxFields(200)(         // query analyzer that limit query fields
```

Comment:
Unrelated: But looking at this could probably allow optimization by providing some `BooleanAlgebra` to be able to do: `maxDepth(30) || maxFields(200)`

Reply:
And/Or operations make sense for the 2 examples here but `QueryAnalyzer` is more general than that and allows any arbitrary effect to run and possibly modify the query, so I'm not sure it makes in a general context.

I like the `@@` notation though, not only is it consistent with ZIO Test, but it also reduce nesting, it's more discoverable and doesn't require me to duplicate every query analyzer / wrapper like I'm doing now. The only problematic one is `apolloTracing` because its construction is effectful so it can't be stacked with others...

Reply:
Maybe I could create a new wrapper type `CombinedWrapper` that takes an optional effectful state and can contain multiple wrappers. That way I'll be able to express `apolloTracing` as a wrapper. Will try that.